### PR TITLE
Renamed scripts folder to autogpt. Absolute Imports. isort. Fixed flake8 F401, E402

### PR DESCRIPTION
**Pull request by ryanpeach**
https://github.com/Torantulino/Auto-GPT/issues/137

### CODE REVIEW

**Review by yousefissa — DISMISSED** [review]

**Comment by ryanpeach** [comment]
Fixed merge conflicts.

**Comment by coleyr** [comment]
I like the PR, saw this on a callout issue and in discord. There are many PR's moving to this common organization.

File: README.md [hunk 1]
```diff
@@ -84,32 +87,67 @@ git clone https://github.com/Torantulino/Auto-GPT.git
...
-1. Run the `main.py` Python script in your terminal:
+1. Run the `autogpt` Python module in your terminal:
 *(Type this into your CMD window)*
 ```
-python scripts/main.py
+<<<<<<< HEAD
+PYTHONPATH=. python autogpt/main.py
+=======
+python -m autogpt
+>>>>>>> autogpt-namespace-fix-imports
```
Comment (taylor-ennen) [inline README.md]: What is intended to be here? Believe this is from the multi pull requests?

Comment (dhensen) [reply README.md]: After mergin in my PR, you now have __main__.py which should replace main.py, but I see that in this PR main.py still exists.

```
autogpt/__init__.py
autogpt/__main__.py
```
This structure makes it possible to use `python -m autogpt`
The __init__.py makes autogpt folder a proper python module.
The __main__.py makes it the default entrypoint for a command line interface program. (See: https://docs.python.org/3/library/__main__.html#main-py-in-python-packages)

I recommend using `python -m autogpt` and removing main.py.

This is the way.

Comment (ryanpeach) [reply README.md]: > What is intended to be here? Believe this is from the multi pull requests?

Just a mistake, will fix now.

Comment (ryanpeach) [reply README.md]: @dhensen resolved
@taylor-ennen resolved

File: autogpt/commands.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def execute_command(command_name, arguments):
     memory = PineconeMemory()
     try:
```
Comment (taylor-ennen) [inline autogpt/commands.py:58]: this whole block would do well to be put in a command pattern

**Review by taylor-ennen — COMMENTED** [review]
Left a comment on probably some of the next major smells in the code here.

Nothing needs to be fixed though that I see.

Look good to me. Great consolidation @ryanpeach

**Comment by Torantulino** [comment]
Looking to merge this next @ryanpeach.
Are you able to fix the conflicts?

**Comment by nponeccop** [comment]
With such a big PRs, it's better to do in small steps. Can you just make a separate PR with the trailing whitespace fixes, with neither linting nor other things? Then we merge, and you rebase this one on top. Then do Docker fixes. Separately. Rebase this one. Rinse and repeat. And finally when there are no linting-related changes to code anymore, introduce linting in CI as preserving the current state.

File: .github/workflows/linting.yml
Comment (nponeccop) [inline .github/workflows/linting.yml:1]: Linting should be enabled apart, after all the fixes are applied to the existing codebase in separate PRs.

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: Personally I'm against giant `.gitignores` nobody needs. But IMO it's a matter of taste.

File: Dockerfile
Comment (nponeccop) [inline Dockerfile:1]: The requirements fixes are in other PRs already

File: README.md
Comment (nponeccop) [inline README.md:1]: Readme updates deserve a separate PR, especially where they are unrelated to the main idea of this PR

File: README.md [hunk 3]
```diff
@@ -141,6 +170,40 @@ export CUSTOM_SEARCH_ENGINE_ID="YOUR_CUSTOM_SEARCH_ENGINE_ID"
 
 ```
 
+## Redis Setup
```
Comment (nponeccop) [inline README.md]: I mean things like this. Redis setup is clearly unrelated

File: README.md [hunk 4]
```diff
@@ -218,12 +280,11 @@ As an autonomous experiment, Auto-GPT may generate content or take actions that
 
 By using Auto-GPT, you agree to indemnify, defend, and hold harmless the developers, contributors, and any affiliated parties from and against any and all claims, damages, losses, liabilities, costs, and expenses (including reasonable attorneys' fees) arising from your use of this software or your violation of these terms.
 
-## 🐦 Connect with Us on Twitter
```
Comment (nponeccop) [inline README.md:221]: The trailing space fix is both unrelated and in #590

**Comment by ryanpeach** [comment]
@Torantulino I've resolved the conflicts, I'll also make sure the lining workflow passes and I've checked the code.

@nponeccop I completely agree. This needs to be merged after https://github.com/Torantulino/Auto-GPT/pull/142 off which it is branched.

**Review by nponeccop — COMMENTED** [review]
I will leave it as a comment for @Torantulino to consider. I'm not a maintainer, he is who decides on the merging.

File: README.md (hunk 4 above)
Comment (ryanpeach) [reply README.md:221]: Please comment on https://github.com/Torantulino/Auto-GPT/pull/142

File: README.md (hunk 3 above)
Comment (ryanpeach) [reply README.md:173]: Please comment on https://github.com/Torantulino/Auto-GPT/pull/142

**Comment by ryanpeach** [comment]
@Torantulino I've resolved the conflicts, I'll also make sure the lining workflow passes and I've checked the code.

**Comment by nponeccop** [comment]
Linting failed

File: .gitignore
Comment (ryanpeach) [reply .gitignore:1]: This came from one of the merged PR's but I believe .gitignores should be comprehensive. Github itself seems to agree

https://github.com/github/gitignore/blob/main/Python.gitignore

File: .github/workflows/linting.yml
Comment (ryanpeach) [reply .github/workflows/linting.yml:1]: I can agree with that going in another PR.

File: README.md (hunk 3 above)
Comment (ryanpeach) [reply README.md]: Idk how this snuck in to be honest, thanks for the catch.

File: README.md
Comment (ryanpeach) [reply README.md:1]: Some of the readme updates are related to the `autogpt` name change. Some came from the other PR's.

**Comment by ryanpeach** [comment]
So for note I agree with @nponeccop that this PR got away from us. It was slightly political due to the competing PR's of https://github.com/Torantulino/Auto-GPT/pull/329 https://github.com/Torantulino/Auto-GPT/pull/444 .

Due to the limited time in @Torantulino 's day, and the general benefit of all the PR's merged in this one, and the lack of focus of any of the original PR's, and the priority of these changes, I vote to just move it through. If we could pick up speed on code review and merging, I think separate PR's are the way to go.

However, until we make a module out of the code, it's basically impossible to do work in the repo.

**Comment by ryanpeach** [comment]
@nponeccop @Torantulino this is the best I can do on an isolated PR. Your decision https://github.com/Torantulino/Auto-GPT/pull/621

**Comment by nponeccop** [comment]
I guess this one should be changed to a draft PR now.

**Comment by ryanpeach** [comment]
Maintaining a fork until basic code quality is reached and a team of PR reviewers is elected.
